torch/distributions/utils.py

```diff
@@ -130,19 +131,34 @@ def probs_to_logits(probs, is_binary=False):
     return torch.log(ps_clamped)
 
 
-class lazy_property:
+T = TypeVar("T", covariant=True)
+
+
+class lazy_property(Generic[T]):
     r"""
     Used as a decorator for lazy loading of class attributes. This uses a
     non-data descriptor that calls the wrapped method to compute the property on
     first call; thereafter replacing the wrapped method into an instance
     attribute.
     """
 
-    def __init__(self, wrapped):
-        self.wrapped = wrapped
+    def __init__(self, wrapped: Callable[..., T]) -> None:
+        self.wrapped: Callable[..., T] = wrapped
```

Comment:
```suggestion
    def __init__(self, wrapped: Callable[P, T]) -> None:
        self.wrapped: Callable[P, T] = wrapped
```
where P is a typing_extensions.ParamSpec, we don't want to throw away the input arg typing information. Note fixing this may cause errors elsewhere in mypy since the type information will actually propogate now, that is fine.